### PR TITLE
fix a bunch of fetch tests on windows

### PR DESCRIPTION
**Pull request by gvilums**
### What does this PR do?
Updates a bunch of the fetch tests to connect to `localhost` on windows instead of `0.0.0.0` (which is invalid).

Also updates a `chmod` test to be skipped on windows.

### How did you verify your code works?
The tests now pass.

### CODE REVIEW

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
The bug is that `server.hostname` is wrong, not the tests

**Comment by Jarred-Sumner** [comment]
We shouldn't return a hostname which cannot be connected to

File: test/js/web/fetch/fetch.stream.test.ts [hunk 1]
```diff
@@ -55,7 +58,7 @@ describe("fetch() with streaming", () => {
           },
         });
 
-        const server_url = `http://${server.hostname}:${server.port}`;
+        const server_url = `http://${connect_host}:${server.port}`;
```
Comment (cirospaciari) [inline test/js/web/fetch/fetch.stream.test.ts:61]: if it is a `Bun.serve` we can use `server.url.href` so IPv6 will be formated for url

image

**Comment by gvilums** [comment]
Closed in favor of #8770

**Pull request closed without merging**